Homework-6/Controllers/StudentController.cs

```diff
@@ -0,0 +1,160 @@
+using Homework_6.Model;
+using Homework_6.Service;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Homework_6.Controllers
+{
+    /// <summary>
+    /// Controller for managing student data.
+    /// </summary>
+    [ApiController]
+    [Route("api/[controller]")]
+    public class StudentController : ControllerBase
+    {
+        private readonly StudentService _studentService;
+        private readonly ILogger<StudentController> _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StudentController"/> class.
+        /// </summary>
+        /// <param name="studentService">The student service.</param>
+        /// <param name="logger">The logger.</param>
+        public StudentController(StudentService studentService, ILogger<StudentController> logger)
+        {
+            _studentService = studentService;
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Adds a new student.
+        /// </summary>
+        /// <param name="student">The student to add.</param>
+        /// <returns>The ID of the added student.</returns>
+        [HttpPost]
+        public IActionResult AddStudent([FromBody] StudentRequestDTO student)
+        {
+            try
```

Comment:
Create a global exception handler and use try catch inside controller only if you want a specific logic on error.